### PR TITLE
Add LoadOrQuitAction

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I'm not completely sure why this has to be a game action. But I'm not going to rock the boat.
This command isn't networked so I'm not sure if there is a point to doing the serialisation method. Maybe useful for replay?

### CODE REVIEW

**Comment by ZehMatt** [comment]
Yeah for replays to work you need to serialise the data but I highly doubt this should be replayed to begin with. Well for server logging perhaps?

File: src/openrct2/actions/LoadOrQuitAction.hpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    OpenSavePrompt,
+    CloseSavePrompt
+};
+
+DEFINE_GAME_ACTION(LoadOrQuitAction, GAME_COMMAND_LOAD_OR_QUIT, GameActionResult)
+{
+private:
+    uint8_t _mode{ 0 };
+    uint8_t _savePromptMode{ 0 };
```
Comment (ZehMatt) [inline src/openrct2/actions/LoadOrQuitAction.hpp]: I think this should be a bool.

File: src/openrct2/actions/LoadOrQuitAction.hpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
```
Comment (ZehMatt) [inline src/openrct2/actions/LoadOrQuitAction.hpp]: 2019

File: src/openrct2/actions/LoadOrQuitAction.hpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/openrct2/actions/LoadOrQuitAction.hpp]: Turns out its of type enum `PROMPT_MODE` so I've kept it as `uint8_t` and add the appropriate constants.

Comment (ZehMatt) [reply src/openrct2/actions/LoadOrQuitAction.hpp]: I see, since it was either 0 or 1 I sort of assumed its nothing more than that.

**Comment by ZehMatt** [comment]
Do we need for CLIENT_ONLY a new network version? The server doesn't care about this action.

**Comment by duncanspumpkin** [comment]
No probably don't need to make it a network increment

**Comment by Gymnasiast** [comment]
The code suggests that the server can send this to the clients, but not the other way round.

**Comment by ZehMatt** [comment]
Can you point at the given code? CLIENT_ONLY will only act locally.

**Comment by Gymnasiast** [comment]
@ZehMatt Meh, ignore what I said, it was just what I gathered from what the old code did.

Has this been tested with a server and client, closing the server down, and closing the client down?

**Comment by duncanspumpkin** [comment]
Yes its tested on both.

**Pull request merged**